repos/system_upgrade/common/actors/checkmicroarchitecture/libraries/checkmicroarchitecture.py

```diff
@@ -0,0 +1,77 @@
+
+
+def _inhibit_upgrade(missing_flags, target_rhel, microarch_ver, extra_report_fields=None):
+    title = 'Current x86-64 microarchitecture is unsupported in {0}'.format(target_rhel)
+    summary = ('{0} has a higher CPU requirement than older versions, it now requires a CPU '
+               'compatible with {1} instruction set or higher.\n\n'
+               'Missings flags detected are: {2}\n').format(target_rhel, microarch_ver, ', '.join(missing_flags))
+
+    report_fields = [
+        reporting.Title(title),
+        reporting.Summary(summary),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Groups([reporting.Groups.INHIBITOR]),
+        reporting.Groups([reporting.Groups.SANITY]),
+        reporting.Remediation(hint=('If a case of using virtualization, virtualization platforms often allow '
+                                    'configuring a minimum denominator CPU model for compatibility when migrating '
+                                    'between different CPU models. Ensure that minimum requirements are not below '
+                                    'that of {0}\n').format(target_rhel)),
+    ]
+
+    if extra_report_fields:
+        report_fields += extra_report_fields
+
+    reporting.create_report(report_fields)
+
+
+def process():
+    """
+    Check whether the processor matches the required microarchitecture.
+    """
+
+    if not matches_architecture(ARCH_X86_64):
+        api.current_logger().info('Architecture not x86-64. Skipping microarchitecture test.')
+        return
+
+    cpuinfo = next(api.consume(CPUInfo))
+
+    rhel9_microarch_article = reporting.ExternalLink(
+        title='Building Red Hat Enterprise Linux 9 for the x86-64-v2 microarchitecture level',
+        url='https://red.ht/rhel-9-intel-microarchitectures'
+    )
+
+    rhel_major_to_microarch_reqs = {
+        '9': MicroarchInfo(microarch_ver='x86-64-v2',
+                           required_flags=(X86_64_BASELINE_FLAGS + X86_64_V2_FLAGS),
+                           extra_report_fields=[rhel9_microarch_article]),
+        '10': MicroarchInfo(microarch_ver='x86-64-v3',
+                            required_flags=(X86_64_BASELINE_FLAGS + X86_64_V2_FLAGS + X86_64_V3_FLAGS),
+                            extra_report_fields=[]),
+    }
+
+    microarch_info = rhel_major_to_microarch_reqs.get(get_target_major_version())
+    if not microarch_info:
+        api.current_logger().info('No known microarchitecture requirements are known for target RHEL%s.',
+                                  get_target_major_version())
+        return
+
+    missing_flags = [flag for flag in microarch_info.required_flags if flag not in cpuinfo.flags]
+    api.current_logger().debug('Required flags missing: %s', missing_flags)
+    if missing_flags:
+        _inhibit_upgrade(missing_flags,
+                         'RHEL{0}'.format(get_target_major_version()),
```

Comment:
nitpick `RHEL9` -> `RHEL 9`
```suggestion
                         'RHEL {0}'.format(get_target_major_version()),
```

Reply:
@MichalHe  wait a sec with the change.